modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/clients/APIPublisherRestClient.java

```diff
@@ -380,4 +382,67 @@ public HttpResponse createNewAPI(String provider, String apiName, String version
 			throw new Exception("API Subscription failed> " + response.getData());
 		}
 	}
+
+	/**
+	 * Change the API Lifecycle status to Publish with the option of Re-subscription is required or not
+	 *
+	 * @param apiIdentifier           - Instance of APIIdentifier
+	 * @param isRequireReSubscription - true if Re-subscription is required else false.
+	 * @return HttpResponse - Response of the API publish event
+	 * @throws Exception - Exception Throws in checkAuthentication() and when do the REST service calls to do the
+	 *                   lifecycle change
+	 */
+	public HttpResponse changeAPILifeCycleStatusToPublish(APIIdentifier apiIdentifier, boolean isRequireReSubscription)
+			throws Exception {
+		checkAuthentication();
+		APILifeCycleStateRequest publishUpdateRequest = new APILifeCycleStateRequest(apiIdentifier.getApiName(),
+				apiIdentifier.getProviderName(),
+				APILifeCycleState.PUBLISHED);
+		publishUpdateRequest.setVersion(apiIdentifier.getVersion());
+		String requestParameters = publishUpdateRequest.generateRequestParameters();
+		if (isRequireReSubscription) {
+			requestParameters += "&requireResubscription=true";
+		}
+
+		HttpResponse response = HttpRequestUtil.doPost(new URL(backEndUrl +
+				"/publisher/site/blocks/life-cycles/ajax/life-cycles.jag")
+				, requestParameters
+				, requestHeaders);
+		if (response.getResponseCode() == 200) {
```

Comment:
Don't hard code the response code. You can refer response codes constants/enum from http client libraries.